reframe/core/buildsystems.py

```diff
@@ -881,6 +887,10 @@ def emit_build_commands(self, environ):
             ret.append(f'spack -e {self.environment} config add '
                        f'"config:install_tree:root:{install_tree}"')
```

Comment:
Since you are doing exactly the same later with the `config_opts`, I would instead add an entry in a local `config_opts` that I would then iterate on. Something like this:

```python
config_opts = []
if self._auto_env:
    config_opts.append(...)

config_opts += self.config_opts
for opt in config_opts:
    ... 
```

Reply:
Good call. Is [9721b13](https://github.com/reframe-hpc/reframe/pull/2819/commits/9721b131a0473f1326f27850bfc690572e090189) what you're suggesting?